src/main/java/com/intel/gkl/smithwaterman/IntelSmithWaterman.java

```diff
@@ -95,6 +95,15 @@ public synchronized boolean load(File tempDir) {
     @Override
     public SWNativeAlignerResult align(byte[] refArray, byte[] altArray, SWParameters parameters, SWOverhangStrategy overhangStrategy)
     {
+        if(refArray == null)
+            throw new NullPointerException("Reference data array is null.");
+        if(altArray == null)
+            throw new NullPointerException("Alternate data array is null.");
+        if(parameters == null)
+            throw new NullPointerException("Parameter structure is null.");
+        if(overhangStrategy == null)
+            throw new NullPointerException("OverhangStrategy is null.");
+
         int intStrategy =  getStrategy(overhangStrategy);
         byte[] cigar = new byte[2*Integer.max(refArray.length, altArray.length)];
```

Comment:
@mkuligow  : Please add check for cigar.length > 0, and intStrategy in range [9,12].

Reply:
Thank you @mkuligow Looks good!